Marlin/src/gcode/control/M80_M81.cpp

```diff
@@ -79,11 +79,19 @@ void GcodeSuite::M81() {
 
   print_job_timer.stop();
 
-  #if HAS_FAN
-    #if ENABLED(PROBING_FANS_OFF)
-      thermalManager.fans_paused = false;
-      ZERO(thermalManager.saved_fan_speed);
-    #endif
+  #if BOTH(HAS_FAN, PROBING_FANS_OFF)
+    thermalManager.fans_paused = false;
+    ZERO(thermalManager.saved_fan_speed);
+  #endif
+
+  LCD_MESSAGE_F(MACHINE_NAME " " STR_OFF ".");
+
+  #if ENABLED(POWER_OFF_TIMER)
+    if (parser.seenval('D')) { powerManager.setPowerOffTimer(SEC_TO_MS(parser.value_ushort())); return; }
```

Comment:
This does not allow a combination of the two switches, D will always take precedence. You can't cover a scenario where a possible cooldown is wanted, but a minimum delay is necessary for graceful shutdown.